### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Yazino12**
### In this pull:

**Decorate Person class using Decorator pattern of Ruby**

- Create class `Nameable`
- Inherit **Nameable** from **Person**
- Create a `Decorator` class that inherits **Nameable**
- Create `CapitalizeDecorator` and `TrimmerDecorators` that inherit from the base **Decorator** class
- Modify the **correct_name** method in **CapitalizeDecorator** to _capitalize the output_
- Modify the **correct_name** method in **TrimmerDecorator** to _trim the output to 10 chars_

### CODE REVIEW

File: decorators.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require './decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @component.correct_name.upcase
```
Comment (Deepakdanger) [inline decorators.rb]: - [x] Using `.upcase` will return all the letter in capital but we dont need that here, here we have to use `capitalize`, which will return only the first letter capital. For ex:-

```
name="adam"
name.upcase         # "ADAM"
name.capitalize      # "Adam"
```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Yazino12, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorators.rb (hunk 1 above)
Comment (Yazino12) [reply decorators.rb]: @Deepakdanger Thanks for taking the time and pointing out the issue 😇. I will fix the issue right away 😊

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Yazino12👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

image

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**